dot/network/host.go

```diff
@@ -65,15 +63,27 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 	}
 
 	var externalAddr ma.Multiaddr
-	ip, err := pubip.Get()
-	if err != nil {
-		logger.Errorf("failed to get public IP: %s", err)
-	} else {
-		logger.Debugf("got public IP %s", ip)
+	if cfg.PublicIP != "" {
+		ip := net.ParseIP(cfg.PublicIP)
+		if ip == nil {
+			return nil, fmt.Errorf("invalid public ip: %s", cfg.PublicIP)
```

Comment:
Define an error at global scope e.g. `var ErrPublicIPIsNotValid = errors.New("...")` and have this as

```suggestion
			return nil, fmt.Errorf("%w: %s", ErrPublicIPIsNotValid, cfg.PublicIP)
```

Reply:
The other errors for the host are produced with `fmt.Errorf`.  I don't mind doing this, but we should be changing all the errors for the node.

Reply:
Not really, just do it bit by bit I think, especially since it's new code. I don't believe we should go all in and refactor all errors returned :sweat_smile:

Reply:
`host` is a private struct.  Not sure this is really required, since the only thing calling `newHost` is `NewService` and even that doesn't have package errors at this point.